core.js

```diff
@@ -1841,14 +1841,18 @@ exokit.setNativeBindingsModule = nativeBindingsModule => {
      * Wrap AudioContext.DecodeAudioDataSync binding with promises and callbacks.
      */
     decodeAudioData(arrayBuffer, successCallback, errorCallback) {
-      try {
-        let audioBuffer = this._decodeAudioDataSync(arrayBuffer);
-        if (successCallback) { successCallback(audioBuffer); }
-        return Promise.resolve(audioBuffer);
-      } catch(err) {
-        if (errorCallback) { errorCallback(err); }
-        return Promise.reject(err);
-      }
+      return new Promise((resolve, reject) => {
+        process.nextTick(() => {
+          try {
+            let audioBuffer = this._decodeAudioDataSync(arrayBuffer);
+            if (successCallback) { successCallback(audioBuffer); }
```

Comment:
I would `nextTick` only the callbacks, not the whole `Promise`.

Also, this is a case where we're calling into user code from outside of a user stack, so we should `try`-`catch` and perhaps `console.warn` the catch. Otherwise, I think if the user code throws then it will be an uncaught exception and it would be snagged/bring the core down.